### PR TITLE
Fix: Move compliance retention to free

### PR DESCRIPTION
**Pull request by dhrupo**
## What does this PR do and why?

Moves form-level compliance retention into the free plugin by exposing the compliance settings UI, persisting the retention values in free, and adding a free-side scheduled cleanup handler. The implementation keeps compatibility with Pro by avoiding duplicate retention registration when Pro is active.

**Related issue:** N/A

## Scope

- [x] Free plugin
- [ ] Pro plugin

## Changes

- [x] PHP — app/Services/Compliance/ComplianceService.php: adds scheduled cleanup for auto_delete_days and attachment cleanup for the free retention flow
- [x] PHP — app/Hooks/actions.php, app/Models/Form.php, app/Modules/Form/Form.php, app/Modules/Form/Settings/FormSettings.php, app/Services/Settings/SettingsService.php: wires the feature into free and aligns legacy/default settings paths
- [x] Vue — resources/assets/admin/components/settings/FormSettings.vue: enables compliance retention controls in free and initializes the related state keys

## How to test

1. In the free plugin, open a form settings page and enable Compliance Settings with auto delete old entries.
2. Save settings and confirm the values persist after reload.
3. Create entries older than the configured delay and run the scheduled retention hook.
4. Verify old entries and uploaded files for those entries are removed in free-only mode.
5. Activate Pro and verify the same form still uses Pro handling without duplicate cleanup behavior.

## Anything the reviewer should know?

- npm run production passed during verification.
- The free repo commit does not include vendor autoload artifacts or unrelated build output.

### CODE REVIEW

**Review by autharif — CHANGES_REQUESTED** [review]
<h3>Summary</h3>
<p>The change is narrowly scoped, but there is one blocker to address before merge.</p>
<p><strong>Key changes:</strong></p>
<ul>
<li><code>app/Hooks/Ajax.php</code> keeps the general-settings save action behind <code>Acl::verify(&#39;fluentform_forms_manager&#39;)</code>, and <code>app/Modules/Acl/Acl.php</code> shows that verification also enforces the admin nonce check before permission evaluation.</li>
<li><code>app/Modules/Form/Settings/FormSettings.php</code> and <code>app/Services/Settings/SettingsService.php</code> both sanitize the new retention fields before persistence, with <code>auto_delete_days</code> normalized to an integer and the meta removed when the feature is turned off.</li>
</ul>
<p><strong>Issues found:</strong></p>
<ul>
<li>retention batch cap (<code>app/Services/Compliance/ComplianceService.php:61:66</code>)</li>
</ul>
<h3>Confidence Score: 3/5</h3>
<ul>
<li>Merge stance: <code>REQUEST CHANGES</code> with 1 confirmed blocker-level finding(s).</li>
<li>Verification confirmed 0 Critical and 1 Important finding(s) in the changed files. The security-sensitive save path remains gated by the shared ACL/nonce flow, and the new retention settings are sanitized before they reach storage or the cron cleanup path.</li>
</ul>
<details><summary>Prompt To Fix All With AI</summary>

```text
Apply the confirmed review fixes for fluentform/fluentform#857 at commit a2a36dce3cc5ebffe1245777367127cd51ebfd46.
Keep behavior unchanged outside these issues.

1. retention batch cap
What: The cleanup job only fetches and deletes one batch of 100 expired submissions per delay bucket on each daily run.
Why: On any form that accumulates more than 100 eligible rows between runs, some submissions will remain past the configured retention window until later executions. That undermines the compliance cleanup guarantee this feature is meant to provide.
Fix: Keep chunking until no eligible rows remain, or page by primary key in a loop so the job drains the full expired set instead of stopping after the first 100 rows.
```
</details>
<p><sub>Last reviewed commit: <a href="https://github.com/fluentform/fluentform/commit/a2a36dce3cc5ebffe1245777367127cd51ebfd46">Fix: Move compliance retention to free</a></sub></p>

**Review by autharif — COMMENTED** [review]
Rechecked [`ddcb1c1`](https://github.com/fluentform/fluentform/pull/857/commits/ddcb1c1e224ca340c75d4564d02d3d69d40a49ed). 1 earlier blocker was addressed, but new issues were found.
<p><strong>New Blockers:</strong> New blockers were introduced in this push.</p>
<ul>
<li>auto-delete-meta-needs-unique-key (<code>app/Services/Compliance/ComplianceService.php:21:43</code>)</li>
<li>reparse attachment fields per batch (<code>app/Services/Compliance/ComplianceService.php:90:104</code>)</li>
</ul>
<p><strong>Additional Blocker:</strong> I also found another blocker in related code.</p>
<ul>
<li>retention-flag-type-mismatch (<code>app/Services/Settings/SettingsService.php:96:105</code>)</li>
</ul>

**Review by autharif — COMMENTED** [review]
Rechecked [`c03566d`](https://github.com/fluentform/fluentform/pull/857/commits/c03566ddc4fb5ceabe23bcab34a784a0ba67e975). 3 earlier blockers were addressed, but new issues were found.
<p><strong>New Blocker:</strong> A new blocker was introduced in this push.</p>
<ul>
<li>Migrated retention settings are skipped by cron (<code>app/Services/Compliance/ComplianceService.php:24:28</code>)</li>
</ul>
<p><strong>Additional Blocker:</strong> I also found another blocker in related code.</p>
<ul>
<li>Undefined _ff.merge breaks settings hydration (<code>resources/assets/admin/components/settings/FormSettings.vue:954</code>)</li>
</ul>
<p><strong>Current Open Blockers:</strong> This PR has already gone through multiple blocker reruns, so here is the full blocker set I still see in the latest code.</p>
<ul>
<li>Migrated retention settings are skipped by cron (<code>app/Services/Compliance/ComplianceService.php:24:28</code>)</li>
<li>Undefined _ff.merge breaks settings hydration (<code>resources/assets/admin/components/settings/FormSettings.vue:954</code>)</li>
</ul>

**Review by autharif — COMMENTED** [review]
Rechecked [`addedb2`](https://github.com/fluentform/fluentform/pull/857/commits/addedb2d6cc2a9d647cc075676f2f9636b847eb6). 2 earlier blockers were addressed, but new issues were found.
<p><strong>Additional Blockers:</strong> I also found additional blockers in related code.</p>
<ul>
<li>Retention delete path skips deprecated attachment-delete hook contract (<code>app/Services/Compliance/ComplianceService.php:135</code>)</li>
<li>N+1 submission queries during attachment cleanup (<code>app/Services/Compliance/ComplianceService.php:74:145</code>)</li>
</ul>
<p><strong>Current Open Blockers:</strong> This PR has already gone through multiple blocker reruns, so here is the full blocker set I still see in the latest code.</p>
<ul>
<li>Retention delete path skips deprecated attachment-delete hook contract (<code>app/Services/Compliance/ComplianceService.php:135</code>)</li>
<li>N+1 submission queries during attachment cleanup (<code>app/Services/Compliance/ComplianceService.php:74:145</code>)</li>
</ul>

**Review by autharif — APPROVED** [review]
Rechecked [`0a5f130`](https://github.com/fluentform/fluentform/pull/857/commits/0a5f130ba7cc6f69fba6b938cbb073b8025e77a8). The earlier blockers are addressed, and I do not see any remaining blocker-level issues in the latest changes.
